src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/dependency/IntegrationTest.java

```diff
@@ -0,0 +1,235 @@
+import com.sonymobile.tools.gerrit.gerritevents.dto.events.ChangeBasedEvent;
+import com.sonymobile.tools.gerrit.gerritevents.dto.events.GerritTriggeredEvent;
+import hudson.model.FreeStyleProject;
+import hudson.model.Job;
+import hudson.tasks.Shell;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * The set of integration tests for dependency feature of Gerrit Trigger Plugin.
+ * Can be moved or renamed in future to be more generic.
+ */
+public class IntegrationTest {
+    /**
+     * An instance of Jenkins Rule.
+     */
+    // CS IGNORE VisibilityModifier FOR NEXT 2 LINES. REASON: JenkinsRule.
+    @Rule
+    public final JenkinsRule jenkinsRule = new JenkinsRule();
+
+    private static final int TIMEOUT = 10;
+    private static final int POLLING_INTERVAL = 1000;
+
+    /**
+     * Test that child build see the env variables that contains information about "parent" build.
+     * @throws Exception throw if so
+     */
+    @Test
+    public void testChildBuildSeeParametersOfParentJob() throws Exception {
+        String gerritServerName = "";
+        GerritServer gerritServer = createMockGerritServer(gerritServerName);
+        PluginImpl.getInstance().addServer(gerritServer);
+
+        ChangeBasedEvent gerritEvent = createMockChangeBasedEvent(gerritServerName);
+        List<GerritProject> projectList = createMockGerritProjectList();
+        List<PluginGerritEvent> triggerOnEvents = createMockPluginGerritEventList();
+
+        GerritTrigger parentTrigger = new GerritTrigger(projectList);
+        FreeStyleProject parent = createJobWithGerritTrigger("parent", parentTrigger, triggerOnEvents);
+
+        GerritTrigger childTrigger = new GerritTrigger(projectList);
+        childTrigger.setDependencyJobsNames(parent.getName());
+        FreeStyleProject child = createJobWithGerritTrigger("child", childTrigger, triggerOnEvents);
+        child.getBuildersList().add(new Shell("test \"$DEP_parent_BUILD_NAME\" == \"parent\""));
+
+
+        PluginImpl.getInstance().getHandler().notifyListeners(gerritEvent);
+
+        waitCompletedBuild(parent, TIMEOUT);
+        waitCompletedBuild(child, TIMEOUT);
+
+        jenkinsRule.assertBuildStatusSuccess(parent.getLastCompletedBuild());
+        jenkinsRule.assertBuildStatusSuccess(child.getLastCompletedBuild());
+    }
+
+    /***
+     * asdfasdf.
+     * @throws Exception if so.
+     */
+    @Test
+    public void testChildWorkflowBuildSeeParametersOfParentJob() throws Exception {
```

Comment:
@jglick, @rsandell, @oleg-nenashev this test passed (at least locally), so, should be ok :)